### PR TITLE
changed precondition of locale provider tests so that locale is not i…

### PR DESCRIPTION
**Pull request by tomhoefer**
…nitialized by any other previous test

Signed-off-by: Thomas Höfer <t.hoefer@telekom.de>

### CODE REVIEW

**Comment by tomhoefer** [comment]
fixes #1731

**Comment by kaikreuzer** [comment]
Wouldn't it be better to clear the CM entry instead? You never know, when CM might call the modified() method and your locale=null could be overwritten again.

**Comment by maggu2810** [comment]
> Wouldn't it be better to clear the CM entry instead?

I think this would be the better one.

I started yesterday to add utility functions to the OSGiTest base class.
I dropped that after a few minutes because I am not sure if we want a hard dependency to the configuration admin and the locale provider in that base class.
But at least some function (set locale by CM, clear locale by CM, set locale by calling modified itself, ...) to be reused in different tests could be usefull.

**Comment by tomhoefer** [comment]
One question: How would you clear the locale by CM? Deleting the CM entry is not an option because after that the service is not longer available (have tried all configuration policies). Shall we add an additional property e.g. clearLocale? Or do I miss here something?

**Comment by maggu2810** [comment]
Shouldn't modified be called with an empty configuration if you clear all entries in the CM?
Can't test it ATM, but that the service disappears if the configuration gets lost, seems to be wrong. I mean that shouldn't be the case. Shouldn't it?
It already starts with an empty configuration.
Perhaps I will find some time later.

**Comment by tomhoefer** [comment]
At least if I invoke config.delete() then the service gets lost...

> Shouldn't modified be called with an empty configuration if you clear all entries in the CM?

Do you mean deleting all properties of CM and then update the CM? If yes, then we have to change the implementation of LocaleProvider by setting the locale to null if no language was provided.

**Comment by maggu2810** [comment]
I tested it and it works for me.
Playground: https://github.com/eclipse/smarthome/compare/master...maggu2810:locale-test?expand=1

**Comment by tomhoefer** [comment]
thx... will check your implementation

**Comment by tomhoefer** [comment]
ok... I was not sure if we should set the locale attribute to null if no language is provided... If this is the preferred way to "clear" the locale then I can change my impl...

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java [hunk 1]
```diff
@@ -36,15 +36,16 @@ protected void activate(Map<String, Object> config) {
         modified(config);
     }
 
-    protected void modified(Map<String, Object> config) {
+    protected synchronized void modified(Map<String, Object> config) {
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/LocaleProviderImpl.java]: good catch!

**Pull request merged**